src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs

```diff
@@ -1111,6 +1111,17 @@ public static void Xml_TypeWithMismatchBetweenAttributeAndPropertyType()
         Assert.StrictEqual(value.IntValue, actual.IntValue);
     }
 
+    [Fact]
+    public static void Xml_TypeWithNonPublicDefaultConstructor()
```

Comment:
The test in this PR covers first scenario. The tests for struct such as Xml_Struct would cover second scenario. 
I will add a test to cover the last scenario, which happens when a non-parameterless constructor is declared.